### ISSUE TITLE
inst2dict() into dict2inst() crashes when using constructor parameters

### ISSUE THREAD

**Issue by rokasv** [godotengine/godot#32234]
Godot 3.1.1

### Outline:
A project with this script
```gdscript
extends Node2D

class Temp:
	var value
	func _init(number):
		value = number

func _init():
	var a = Temp.new(1)
	var b = inst2dict(a)
	var c = dict2inst(b) 
```
crashes on `var c` line with the following error:
```diff
-drivers/unix/net_socket_posix.cpp:190 - Socket error: 10054
```
### Reproduction
Open the [zipped project](https://github.com/godotengine/godot/files/3638705/Temp.zip) and try running it.

### Important
The crash only happens if the functions are called on a class with constructor parameters. Specifically, using the following script does not result in a crash:
```gdscript
extends Node2D

class Temp:
	var value
	func _init():
		value = 3

func _init():
	var a = Temp.new()
	var b = inst2dict(a)
	var c = dict2inst(b) 
```

### Tests
1. Adding `tool` keyword to the script crashes the editor on saving 
2. If the script file used is a "Built-In Script", then the program does not crash. Instead, it produces:
```diff
-Error calling built-in function `inst2dict`: Not based on a resource file
```
3. Contrary to #30077, the crash still occurs if the Temp implementation is moved to a loaded script as follows: 
```gdscript
extends Node2D

func _init():
	var a = load("res://Temp.gd").new(1)
	add_child(a)
	var b = inst2dict(a)
	var c = dict2inst(b)
``` 
where Temp.gd is:
```gdscript
extends Node2D

var value
func _init(number):
	value = number
```

### Summary
In other words, dict2inst inst2dict pair cannot be used on classes with custom constructor parameters.

Calling `dict2inst( inst2dict( Class.new(1) ) )` will crash the program.
Calling `dict2inst( inst2dict( Class.new() ) )` will *not* crash the program.

### Edit: 
Adding default values to the constructor parameters also dodges the crashing. I'd guess `dict2inst` is trying to initialize the relevant class without any constructor parameters, which then fail if they are necessary.

**Comment by KoBeWi** [comment]
Duplicate of #30572
